AK/Variant.h

```diff
@@ -40,47 +39,134 @@ consteval IndexType index_of()
     return VariantIndexOf<T, IndexType, 0, Ts...> {}();
 }
 
+template<typename... T>
+union VariantStorage;
+
+template<typename T, typename... Rest>
+union VariantStorage<T, Rest...> {
+    VariantStorage() = default;
+    VariantStorage()
+    requires(!IsTriviallyConstructible<T> || (!IsTriviallyConstructible<Rest> || ...)) {};
+
+    VariantStorage(VariantStorage&&) = default;
+    VariantStorage(VariantStorage const&) = default;
+
+    VariantStorage& operator=(VariantStorage&&) = default;
+    VariantStorage& operator=(VariantStorage const&) = default;
+
+    // Note: Actual destruction is handled by the Variant itself
+    ~VariantStorage() = default;
+    ~VariantStorage()
+    requires(!IsTriviallyDestructible<T> || !(IsTriviallyDestructible<Rest> && ...)) {};
```

Comment:
Nit: maybe do `(!IsTriviallyDestructible<Rest> ||  ...)` (as it is done above) to not needlessly confuse potential readers. (This definitely confused me for a few seconds)